jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -3802,7 +3802,11 @@ public Method run() throws Exception {
 			};
 			
 			Method values = AccessController.doPrivileged(privilegedAction);
-			enums = (T[])values.invoke(this);
+			Object rawEnums = values.invoke(this);
+			if (!rawEnums.getClass().isArray()) {
```

Comment:
Perhaps this should also check `rawEnums == null`:
```java
			if ((rawEnums == null) || !rawEnums.getClass().isArray()) {
```

Reply:
Updated.